sale_order_partner_no_autofollow/tests/test_sale_order_autofollow_disabled.py

```diff
@@ -0,0 +1,152 @@
+from odoo.tests import tagged
+from odoo.tests.common import Form, TransactionCase
+
+
+@tagged("post_install", "-at_install", "standart")
+class TestSaleOrderNoAutofollow(TransactionCase):
+    def setUp(self, *args, **kwargs):
+        """
+        Test for orders, to check autofollow when the
+        'customer no autofollow' mode is enabled in settings
+        """
+        super(TestSaleOrderNoAutofollow, self).setUp(*args, **kwargs)
+
+        self.partner1 = self.env["res.partner"].create(
+            {"name": "Test1", "email": "test1@test.com"}
+        )
+        self.partner2 = self.env["res.partner"].create(
+            {"name": "Test2", "email": "test2@test.com"}
+        )
+        self.product1 = self.env["product.product"].create({"name": "desktop"})
+        self.product2 = self.env["product.product"].create({"name": "table"})
+
+        self.sale_order_1 = self.env["sale.order"].create(
+            {"partner_id": self.partner1.id}
+        )
+        self.env["ir.config_parameter"].sudo().set_param(
+            "sale_order.customer_no_autofollow", True
+        )
+
+    def test_partner_disable_autofollow(self):
+        """
+        'Customer no autofollow' mode is enabled in settings.
+        Test whether the option to disable autofollow is enabled
+        or disabled
+        """
+        self.assertEqual(
+            self.sale_order_1._partner_disable_autofollow(),
+            True,
+            "Must be equal to True",
+        )
+
+    def test_message_subscribe(self):
+        """'Customer no autofollow' mode is enabled in settings.
+            Test whether the user will be added to the autofollow
+        """
+        sale_order = self.sale_order_1
+        sale_order.with_context(
+            sale_partner_disable_autofollow=sale_order._partner_disable_autofollow()
+        ).message_subscribe(partner_ids=[self.partner1.id])
+        self.assertNotIn(
+            sale_order.partner_id.id,
+            sale_order.message_follower_ids.mapped("partner_id").ids,
+            msg="The customer must not be among the subscribers",
+        )
+
+    def test_sale_order_create(self):
```

Comment:
This test makes no sense.
1. You are creating Sales Order in L23 before you are disabling auto follow in L26. So it is created with auto follow enabled NOT disabled.
2. You are adding some lines (why?). This has nothing to do with either create() and followers.

Reply:
Got it, I'll fix it